referenceframe/transformable.go

```diff
@@ -215,15 +215,33 @@ func (gF *GeometriesInFrame) Geometries() []spatialmath.Geometry {
 // GeometryByName returns the named geometry if it exists in the GeometriesInFrame, and nil otherwise.
 // If multiple geometries exist with identical names one will be chosen at random.
 func (gF *GeometriesInFrame) GeometryByName(name string) spatialmath.Geometry {
-	if gF.geometriesMap == nil {
+	if gF.nameIndexMap == nil {
 		return nil
 	}
-	if geom, ok := gF.geometriesMap[name]; ok {
-		return geom
+	if i, ok := gF.nameIndexMap[name]; ok {
+		return gF.geometries[i]
 	}
 	return nil
 }
 
+// RemoveGeometry removes the named geometry if it exists in the GeometriesInFrame and has no effect otherwise
+func (gF *GeometriesInFrame) RemoveGeometry(name string) {
```

Comment:
I don't think we ever want to be doing this; this introduces weird state-y side effects to things, where if a single pointer to a geometries in frame is passed around, something else can modify the object and affect other things.

If this is really really necessary then this should return a NEW GeometriesInFrame that lacks the specified geometry.

Reply:
I initially did not want to do this because the deep copy would be expensive, but IIRC this should only be happening once per plan, so the overhead is probably negligible for the safety

Reply:
I think the way to do this is to remove this function entirely, and instead loop through the original GiF in the collision constraint, make an array of geometries skipping the ones that match, and call NewGeometriesInFrame on that.

Reply:
I like this last idea the most, will do that

Reply:
I have *a* solution for this now, that is not ideal at the moment.  Let me know what you think

Reply:
Yes I had intended to remove, but forgot to